projects/step-core/src/lib/services/plan-dialogs.service.ts

```diff
@@ -40,7 +40,7 @@ export class PlanDialogsService implements PlanLinkDialogService {
   selectPlan(): Observable<Plan> {
     const selectedEntity$ = this._entityDialogs.selectEntityOfType('plans', true);
     const plan$ = selectedEntity$.pipe(
-      map((result) => result.item),
+      map((result) => (result.item as Plan).id!),
       switchMap((id) => this._plansApiService.getPlanById(id))
```

Comment:
What about to do this in the same way for keywords?
```suggestion
      map((result) => result.item as Plan),
      switchMap((plan) => this._plansApiService.getPlanById(plan.id!))
```